src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateWorkitemProcedureStepState.sql

```diff
@@ -0,0 +1,122 @@
+﻿/*************************************************************
+ Stored procedure for Updating a workitem procedure step state.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     UpdateWorkitemProcedureStepState
+--
+-- DESCRIPTION
+--     Updates a UPS-RS workitem procedure step state.
+--
+-- PARAMETERS
+--     @workitemKey
+--         * The Workitem Key.
+--     @procedureStepStateTagPath
+--          * The Procedure Step State Tag Path
+--     @procedureStepState
+--          * The New Procedure Step State Value
+--     @watermark
+--          * The Workitem Watermark
+--     @proposedWatermark
+--          * The Proposed Watermark for the Workitem
+--
+-- RETURN VALUE
+--     The Number of records affected in ExtendedQueryTagString table
+--
+-- EXAMPLE
+--
+/*
+
+        BEGIN
+	        DECLARE @workitemKey BIGINT = 1
+	        DECLARE @procedureStepStateTagPath varchar(64) = '00741000'
+	        DECLARE @procedureStepState varchar(64) = 'SCHEDULED'
+	        DECLARE @watermark BIGINT = 101
+            DECLARE @proposedWatermark BIGINT = 201
+
+	        EXECUTE [dbo].[UpdateWorkitemProcedureStepState] 
+	           @workitemKey
+	          ,@procedureStepStateTagPath
+	          ,@procedureStepState
+              ,@watermark
+	          ,@proposedWatermark
+        END
+        GO
+
+*/
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.UpdateWorkitemProcedureStepState
```

Comment:
Could you quickly validate this sproc and attach the sql query plan. I just want to make sure we have all the indexes correctly placed.

Reply:
[UpdateProcedureStepState-SqlPlan.zip](https://github.com/microsoft/dicom-server/files/8199645/UpdateProcedureStepState-SqlPlan.zip)